lib/std/system/stringimpl.nim

```diff
@@ -29,7 +29,7 @@ proc `=destroy`*(s: string) {.exportc: "nimStrDestroy", inline.} =
   if isAllocated(s): dealloc(s.a)
 
 template safeCopyMem(dest: var string; src: string; len, allocated: int) =
-  if dest.a != nil:
+  if dest.a != StrData(nil):
```

Comment:
The compiler must accept this without the type conversion.

Reply:
Already does apparently, oops